File: solutions/7kyu/Valid_Spacing/valid_spacing.js
function validSpacing(s) {
  return s == s.split(" ").filter(word => word.length > 0).join(" ")
}

console.log(validSpacing('Hello World'))// #= true
console.log(validSpacing(' Hello world'))// #= false
console.log(validSpacing('Hello  world '))// #= false
console.log(validSpacing('cod ewars '))// #= false
console.log(validSpacing( 'Hello'))// #= true
console.log(validSpacing('Helloworld'))// #= true
console.log(validSpacing(' '))// #= false
console.log(validSpacing(''))// #= true

File: solutions/7kyu/Valid_Spacing/valid_spacing.rb
def valid_spacing(s)
  s.match(/\s+\z/) || s.match(/\A\s+/) || s.match(/\s{2,}/) ?  false : true
end
p valid_spacing('Hello World') #= true
p valid_spacing(' Hello world') #= false
p valid_spacing('Hello  world ') #= false
p valid_spacing('cod ewars ') #= false
p valid_spacing( 'Hello') #= true
p valid_spacing('Helloworld') #= true
